pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -302,7 +316,59 @@ func (c *Controller) onServiceEvent(curr interface{}, event model.Event) error {
 }
 
 func (c *Controller) onNodeEvent(obj interface{}, ev model.Event) error {
-	return c.checkReadyForEvents()
+	if err := c.checkReadyForEvents(); err != nil {
+		return err
+	}
+
+	var k8sNodes []*kubernetesNode
+	var extaddresses []string
+	rawNodes := c.nodes.GetStore().List()
```

Comment:
I have changed the nodes informer, to reduce memory cost, it now only include the node metadata.

Reply:
Oh!! that means I have to bring it back?